### PR TITLE
Improved Carousel

### PR DESCRIPTION
**Pull request by jstnjns**
## Overview
The existing carousel isn't so flexible.  Instead of spending a lot of effort to refactor it, I've created a new one with documentation on it's different modes.  The goal is to be able to replicate all the sliders we use throughout the site: home/hero, category slider, and anywhere else we might use it.

Additionally, I've added a component to connect carousels (ref/asForNav implementation of `react-slick`)

## Risks
Low.  The old carousel is still being used in `masterclass`, but all that needs to happen there is update the reference from

```
import { Carousel } from 'mc-components'
```

to

```
import { LegacyCarousel } from 'mc-components'
```

for a drop in replacement.  But it is advised to just update to the new carousel implementation.

## Changes
screen shot 2018-08-30 at 1 34 35 am
screen shot 2018-08-30 at 1 34 46 am

## Issue
#154
#114

### CODE REVIEW

File: src/components/Button/index.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ const Button = ({
   children,
   className,
   onClick,
+  primary,
```
Comment (bernabe9) [inline src/components/Button/index.js]: why did you add this prop again?

Comment (jstnjns) [reply src/components/Button/index.js]: ah, good catch.  to get the errors to stop.  it was meant to be temporary and never merged in.  i will fix this now
